### PR TITLE
self-publish cdt-core

### PR DESCRIPTION
**Pull request by mpollmeier**

### CODE REVIEW

File: joern-cli/frontends/c2cpg/eclipse-cdt/pom-extra [hunk 1]
```diff
@@ -0,0 +1,22 @@
+    <url>https://github.com/ShiftLeftSecurity/ghidra/</url>
+    <licenses>
+        <license>
+            <name>Apache-2.0</name>
+            <url>http://www.apache.org/licenses/LICENSE-2.0</url>
+            <distribution>repo</distribution>
+        </license>
+    </licenses>
+    <name>ghidra</name>
```
Comment (max-leuthaeuser) [inline joern-cli/frontends/c2cpg/eclipse-cdt/pom-extra:9]: Well, that one needs and update everywhere.

File: joern-cli/frontends/c2cpg/eclipse-cdt/eclipse-cdt-core-publish.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+mvn gpg:sign-and-deploy-file -Durl=$SONATYPE_URL -DrepositoryId=$REPO_ID -DpomFile=pom.xml -Dclassifier=sources -Dfile=empty.jar
+mvn gpg:sign-and-deploy-file -Durl=$SONATYPE_URL -DrepositoryId=$REPO_ID -DpomFile=pom.xml -Dclassifier=javadoc -Dfile=empty.jar
+mvn gpg:sign-and-deploy-file -Durl=$SONATYPE_URL -DrepositoryId=$REPO_ID -DpomFile=pom.xml -Dfile=$LOCAL_JAR
+
+# remove temporary working artifacts
+rm $LOCAL_JAR pom.xml empty.jar *.asc
+
+echo "artifacts are now published to sonatype staging. next step: log into https://s01.oss.sonatype.org -> staging repositories -> select the right one -> close -> release"
+echo "you can monitor the maven sync status on https://s01.oss.sonatype.org/content/groups/public/io/joern/eclipse-cdt-core/ and https://repo1.maven.org/maven2/io/joern/eclipse-cdt-core/"
+echo "once it's synchronised to maven central (repo1), update the ghidra version in 'joern/joern-cli/frontends/ghidra2cpg/build.sbt'"
```
Comment (max-leuthaeuser) [inline joern-cli/frontends/c2cpg/eclipse-cdt/eclipse-cdt-core-publish.sh:56]: Some 'ghidra' here also.

**Comment by max-leuthaeuser** [comment]
That should not be needed. I made all the stuff public in CDT so no need for shadowing anymore (just move the https://github.com/joernio/joern/blob/master/joern-cli/frontends/c2cpg/src/main/scala/org/eclipse/cdt/internal/core/parser/scanner/MacroArgumentExtractor.scala).

**Comment by mpollmeier** [comment]
looks good now, please re-review

File: joern-cli/frontends/c2cpg/eclipse-cdt/pom-extra [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    <name>ghidra</name>
+    <scm>
+        <url>https://github.com/eclipse-cdt/cdt</url>
+        <connection>scm:git@github.com:eclipse-cdt/cdt.git</connection>
+    </scm>
+    <developers>
+        <developer>
+            <id>max-leuthaeuser</id>
+            <name>Niko Schmidt</name>
```
Comment (maltek) [inline joern-cli/frontends/c2cpg/eclipse-cdt/pom-extra:24]:
```suggestion
    <name>cdt</name>
    <scm>
        <url>https://github.com/eclipse-cdt/cdt</url>
        <connection>scm:git@github.com:eclipse-cdt/cdt.git</connection>
    </scm>
    <developers>
        <developer>
            <id>max-leuthaeuser</id>
            <name>Max Leuthäuser</name>
```

**Review by maltek — APPROVED** [review]

**Pull request merged**